File: record-api/app/cache.py
import aioredis
from app.config import REDIS_URL

redis = None

async def get_redis():
    global redis
    if not redis:
        redis = await aioredis.from_url(REDIS_URL)
    return redis

async def cache_set(key: str, value: str, expire: int = 300):
    r = await get_redis()
    await r.set(key, value, ex=expire)

async def cache_get(key: str):
    r = await get_redis()
    value = await r.get(key)
    if value:
        print("HIT CACHE")
    else:
        print("MISS CACHE")
    return value

File: receive-send-api/src/redis/cache.js
const { createClient } = require('redis');
const client = createClient({ url: process.env.REDIS_URL || 'redis://redis:6379/0' });

client.connect();

async function cacheGet(key) {
  const value = await client.get(key);
  if (value) {
    console.log('HIT CACHE');
    return JSON.parse(value);
  }
  console.log('MISS CACHE');
  return null;
}

async function cacheSet(key, value, expire = 300) {
  await client.set(key, JSON.stringify(value), { EX: expire });
}

module.exports = { cacheGet, cacheSet };